common/processors/follow_leader_processor.py

```diff
@@ -7,17 +7,19 @@
 from common.components.velocity import Velocity
 from common.engine.processor import Processor
 from common.engine.resource_manager import ResourceManager
-from common.resources.time_providers.real_time_provider import RealTimeProvider
+from common.engine.time_providers import RealTimeProvider
 
 logger = logging.getLogger(__name__)
 
 
 class FollowLeaderProcessor(Processor):
     def process(self, r: ResourceManager) -> None:
         time_provider = r.get_resource(RealTimeProvider)
-        for ent, (lead, pos, vel) in esper.get_components(Leader, Position, Velocity):
-            if lead.ent == -1 or not esper.has_component(lead.ent, Position):
+        for ent, (lead, pos, vel) in esper.get_components(Leader, Position,
+                                                          Velocity):
+            if lead.ent == -1:  # no leader defined
                 continue
-            leader_pos = esper.component_for_entity(lead.ent, Position)
-            vel.x = (leader_pos.x - pos.x) * lead.attraction
-            vel.y = (leader_pos.y - pos.y) * lead.attraction
+            leader_pos: Position = esper.component_for_entity(lead.ent,
+                                                              Position)
+            vel.x += (leader_pos.x - pos.x) * lead.attraction
+            vel.y += (leader_pos.y - pos.y) * lead.attraction
```

Comment:
add `__add__` operator to `vel`

Reply:
We can't do that because when we try to modify `vel` rather than `vel.attribute` it create a local instance rather than updating current one